response/slack/cache.py

```diff
@@ -0,0 +1,68 @@
+import logging
+from django.conf import settings
+from django.db import transaction
+
+from response.core.models import ExternalUser, GetOrCreateSlackExternalUser
+from response.slack.client import SlackError
+
+logger = logging.getLogger(__name__)
+
+
+def update_user_cache():
+    cursor = None
+    while True:
+        response = settings.SLACK_CLIENT.get_paginated_users(limit=200, cursor=cursor)
+
+        users = response['members']
+
+        logger.info(f"Updating {len(users)} users in the cache")
+        with transaction.atomic():
+            for user in users:
+                GetOrCreateSlackExternalUser(
+                    external_id=user['id'],
+                    defaults={
+                        'display_name': user['profile']['display_name_normalized'] or user['name'],
+                        'full_name': user["profile"]["real_name"] or user['name'],
+                    },
+                )
+        cursor = response['response_metadata'].get('next_cursor')
+
+        if not cursor:
+            break
+
+
+def get_user_profile(external_id):
```

Comment:
In another PR, we should replace all usages of settings.SLACK_CLIENT.get_user_profile with this to take advantage of the cache.